### PR TITLE
{devel}[GCCcore/7.3.0] Cmake v3.12.0

### PR DESCRIPTION
**Pull request by jhein32**
Cmake easyconfig for version 3.12.0 in GCCcore 7.3.0

### CODE REVIEW

**Comment by boegel** [comment]
@jhein32 Do you have a particular need for this over the existing `CMake-3.11.4-GCCcore-7.3.0.eb`?

If not, I'd prefer sticking to CMake 3.11.4 for the `2018b` toolchain generation...

**Comment by jhein32** [comment]
I prepared this for the Gromacs-Plumed I was working on. Åke has now submitted these as 2 PRs, so I thought I just push this one.

**Comment by pescobar** [comment]
Test report by @pescobar
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
login20.cluster.bc2.ch - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/641a21b16542c3e9be54ef06b77aff44 for a full test report.

**Comment by boegel** [comment]
@jhein32 Unless there's a strong need for a CMake version newer than 3.11.4, I'm included not to include this one, since it will do more 'harm' (cause confusion) than good.

Please let us know if you're OK with this (and if so, close the pull request).

**Comment by jhein32** [comment]
@pescobar Hmm, I am at a loss regarding the build issue you are facing, means I don't understand what went wrong.  On our own system (Haswell, CentOS7) that build like a charm.

@boegel I can live with that one not going in.  I am still at a loss regarding the above error and am wondering whether I (or we) can learn something from this.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2607.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/03ce4489e01a9c64f45e822eab3d608a for a full test report.

**Comment by boegel** [comment]
@jhein32 Not sure what the problem is either, it seems like `ccmake` is being built.

@pescobar Can you provide the full (debug) log?

**Comment by jhein32** [comment]
Hi,

I just discovered that the CMake 3.11.4 is not good enough to support Boost 1.68, while the 3.12.0 does the trick.  I suggest to merge this since CMake is only a build dependency.

Joachim

**Comment by boegel** [comment]
@jhein32 Agreed on that one, see also #6724 (same situation, but different toolchains)

**Comment by boegel** [comment]
@pescobar Can you provide a full debug log so we can check what make the installation for you?

File: easybuild/easyconfigs/c/CMake/CMake-3.12.0-GCCcore-7.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+easyblock = 'ConfigureMake'
+
+name = 'CMake'
+version = '3.12.0'
```
Comment (boegel) [inline easybuild/easyconfigs/c/CMake/CMake-3.12.0-GCCcore-7.3.0.eb]: @jhein32 Do you mind bumping this to `3.12.1`?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2570.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/59c455449d4f83e03b0c4e7e09218d3d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2107.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/8ddc993d431061e4212e14db0e43ae35 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @jhein32!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/5dcbf2795f6f125b6ab4c5a67f464c91 for a full test report.

**Pull request merged**